jazz-build-module/scm-module.groovy

```diff
@@ -0,0 +1,179 @@
+def scm_branch_permission_api_endpoint
+
+def serviceonboarding_repo
+def scm_config
+@Field def scm_protocol	= "http://"
+
+def initialize(configData){
+    scm_config = configData
+    setRepoEndpoints()
+}
+
+def setRepoEndpoints(){
+    scm_user_services_api_endpoint = "${scm_protocol}${scm_config.REPOSITORY.REPO_BASE}/rest/api/1.0/projects/${scm_config.REPOSITORY.REPO_LOC}/repos"
+    scm_core_services_endpoint = "${scm_protocol}${scm_config.REPOSITORY.REPO_BASE}/scm/${scm_config.REPOSITORY.REPO_CORE}/" 
+    scm_user_services_clone_url = "${scm_config.REPOSITORY.REPO_BASE}/scm/${scm_config.REPOSITORY.REPO_LOC}/"
+    scm_webhook_api = "${scm_protocol}${scm_config.REPOSITORY.REPO_BASE}/rest/webhook/1.0/projects/${scm_config.REPOSITORY.REPO_LOC}/repos/"
+    scm_branch_permission_api_endpoint = "${scm_protocol}${scm_config.REPOSITORY.REPO_BASE}/rest/branch-permissions/2.0/projects/${scm_config.REPOSITORY.REPO_LOC}/repos/"
+    serviceonboarding_repo	= "${scm_protocol}${scm_config.REPOSITORY.REPO_BASE}/scm/${scm_config.REPOSITORY.REPO_CORE}/service-onboarding-build-pack.git"
+}
+
+def createProject(repo_owner, repo_name){
+    try { 
+        echo "ready to create project"
+        if(scm_config.JAZZ_SCM == "gitlab") {
+            def git_username
+            def cas_proj_id
+            def cas_repo_id
+            
+            //assuming usernames for now are a concatenation of the user's email, check platform_usermanagement index.js
+            if(repo_owner.contains("@")){
+                git_username = repo_owner.substring(0, owner.indexOf("@"))
+            }
+
+            def user_id = getGitlabUserId(git_username, ${scm_config.GITLAB.gitlab_private_token})
+            
+            def gitlab_repo_output = sh (
+                script: "curl --header \"Private-Token: ${scm_config.GITLAB.gitlab_private_token}\" -X POST \"http://${scm_config.REPOSITORY.REPO_BASE}/api/v3/projects/user/$user_id?name=$repo_name&path=$repo_name&visibility=private&request_access_enabled=true\"",
+                returnStdout: true
+            ).trim()
+
+            def jsonSlurper = new JsonSlurper()
+            def object = jsonSlurper.parseText(gitlab_repo_output)
+            
+            if(object == null || object.equals("") || object.id == null || object.id.equals("")){
+                error "project creation in gitlabs failed"
+            }
+             
+            cas_proj_id = object.id
+
+            cas_repo_id = getCasRepoId(gitlab_private_token, repo_loc)
+
+            transferProject(cas_repo_id, cas_proj_id)
+        }
+        else{
+            sh "curl -X POST -k -v -u \"${scm_config.BITBUCKET.BITBUCKET_USERNAME}:${scm_config.BITBUCKET.BITBUCKET_PASSWORD}\" -H \"Content-Type: application/json\" " + scm_user_services_api_endpoint + " -d \'{\"name\":\""+ repo_name +"\", \"scmId\": \"git\", \"forkable\": \"true\"}\'"
+        }
+    }catch (ex) {
+        if(!((ex.getMessage()).indexOf("groovy.json.internal.LazyMap") > -1)) {
```

Comment:
This logic is not reliable. Use JsonSlurperClassic (if supported) to get ride of lazyMap exceptions

Reply:
Fixed it by having a generic catch as none of the exceptions are actionable.